Update and rename INSTALL to INSTALL.md

gpkvt opened the pull request:
Rewritten and updated ILIAS Installation Howto (for Debian/Ubuntu and RedHat/CentOS)

gpkvt committed:
Update and rename INSTALL to INSTALL.md

gpkvt committed:
Added Contribute section

Incorporated content from proposed README.md (see PR #455)

gpkvt committed:
Fixed typo in TOC

gpkvt committed:
Minor improvements, added Mailing-Lists

gpkvt committed:
Added other dependencies to TOC

Amstutz commented on docs/INSTALL.md:
```diff
@@ -0,0 +1,518 @@
+You can find proper ways to customize ILIAS in the [ILIAS Development Guide](http://www.ilias.de/docu/goto_docu_pg_29964_42.html):
+
+  * [Plugins and Plugin Slots - ILIAS Development Guide](http://www.ilias.de/docu/goto.php?target=st_27029)
+  * [Styles and Templates - ILIAS Development Guide](http://www.ilias.de/docu/goto_docu_pg_210_42.html)
```
Please update to new readme concerning system styles in trunk: [Readme.md](https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/templates/Readme.md)

gpkvt replied:
@Amstutz I've replaced the "Styles and Templates" link with the one you provided.

Amstutz commented:
This is awesome! Thank you very much for all your work.

gpkvt committed:
Updated to new readme concerning system styles

gpkvt committed:
Minor changes on dependencies

Removed nginx from supported webserver list, added experimental db support indication

gpkvt committed:
Added dependency informations for old ILIAS versions

gpkvt committed:
Added support for markdown toc (sublime)

gpkvt committed:
Replaced HTML tables by MD tables

gpkvt committed:
Fixed Typo, added Headlines

gpkvt committed:
Updated to reflect requirements of PR #473

Docs Guideline Proposal

gpkvt committed:
Update INSTALL.md

gpkvt committed:
Minor fixes

gpkvt committed:
Extended instructions on upgrading ILIAS

Added section minor and major updates

tomaszkolonko commented on docs/INSTALL.md:
```diff
@@ -0,0 +1,635 @@
+  * php5-gd, php5-xsl, php5-mysql
+  * OpenJDK 7+
+  * zip, unzip
+
```
Since further down cloning ILIAS is the recommended way, GIT could be mentioned here as well. I am always surprised how many people still don't use git.

tomaszkolonko commented on docs/INSTALL.md:
````diff
@@ -0,0 +1,635 @@
+```
+cd /var/www/html/
+git clone https://github.com/ILIAS-eLearning/ILIAS.git ilias
+git checkout release_5-2
````
cd ilias
before checkout
(for most obvious enough, but I don't know if there is something as "too" obvious)

tomaszkolonko commented on docs/INSTALL.md:
````diff
@@ -0,0 +1,635 @@
+
+On Debian/Ubuntu execute: 
+```
+apt-get install zip unzip php5-gd php5-mysql php5-xslt imagemagick openjdk-7-jdk
````
php5-xslt does not exist
probably: php5-xsl ?

tomaszkolonko commented on docs/INSTALL.md:
```diff
@@ -0,0 +1,635 @@
+## Installation Wizard
+
+After having all dependencies installed an configured you should be able to run the ILIAS Installation Wizard using http://yourservername.org/setup/setup.php
+
```
After having all dependencies installed anD (missing d) configured, restart the apache2 server in order for the packages to take effect. Then you should be able to run.....

If apache2 is not reloaded there are unmet dependencies in the setup. The user installing ILIAS does not get a good error message to pinpoint the problem. Although XLS and GD are both installed ILIAS does not see them.

gpkvt committed:
Updated according to @tomaszkolonko PR comments

gpkvt commented:
@tomaszkolonko, thanks for your input. The changes has been made.

gpkvt committed:
Added informations about the ILIAS testserver

klees commented on docs/INSTALL.md:
```diff
@@ -0,0 +1,656 @@
+| Nginx          | 1.4.6                       |
+| zip            | 3.0                         |
+| unzip          | 6.00                        |
+| JDK            | 1.7.0_121 (IcedTea 2.6.8)   |
```
One last thing: Please add a note that Shibolleth won't work with nginx. Thx!!

gpkvt committed:
Added information about Nginx and Shibboleth

gpkvt committed:
Update INSTALL.md

klees commented:
Dear members of @ILIAS-eLearning/technical-board, are there any problems left? I think everything we talked about in our last meeting is solved now. Please give an approve review.

chfsx commented on docs/INSTALL.md:
```diff
@@ -0,0 +1,658 @@
+
+  * InnoDB storage engine
+  * utf8_general_ci
+  * STRICT_TRANS_TABLES or STRICT_ALL_TABLES disabled (on MySQL 5.6.x)
```
Not sure if a RECOMMEND is enough for the current STRICT_..._TABLES-Situations since a lot of things break without this.

gpkvt replied:
I've changed it to MUST, thanks.

chfsx commented on docs/INSTALL.md:
```diff
@@ -0,0 +1,658 @@
+You may even join us at one of our regular [ILIAS Conferences](http://www.ilias.de/docu/goto.php?target=cat_2255&client_id=docu).
+
+<a name="pull-requests"></a>
+## Pull Requests
```
We describe PR in multiple files, maybe we should reduce it to one and make links.

gpkvt replied:
I'm happy to split this section into several parts or rename it to "Further reading/links" as I think it's a good thing to have the Feature Wiki and Bugtracker mentioned. Is /docs/CONTRIBUTING.md the leading file for informations on pull requests? IIRC I got this part from an (now archived) document on the ILIAS Wiki.

chfsx reviewed (approved):
Thanks a lot for your work! Some small hints from my side but no show-stoppers, merge it!

tomaszkolonko commented:
Dear all, unfortunately I wasn't able to finish walking it through on Monday and plan to do it tomorrow morning (certainly before 11:00 am). I guess there will be only minor high-level suggestions, if any.

gpkvt committed:
Clarify settings for MySQL 5.6

corro commented on docs/INSTALL.md:
```diff
@@ -0,0 +1,659 @@
+For best results we RECOMMEND:
+
+  * Debian GNU Linux 8 / Red Hat Enterprise Linux 7 / Ubuntu 16.04 LTS
+  * MySQL 5.5 / MariaDB
```
I would recommend MySQL >=5.7 as the recommended setup, since 5.5 suffers from multiple known issues:
* Suboptimal InnoDB defaults (such as `innodb_file_per_table` set to `false` by default)
* Suboptimal general defaults (such as `optimizer_search_depth` set way too high by default)
* Bad query optimization in some edge cases (which can cause database lock-ups when using DataCollection)

All these issues were fixed in MySQL 5.7, which is the default version in Ubuntu 16.04 by the way.

gpkvt replied:
Thanks for your comments. I had in mind that there were still some issues with 5.7, but I'm unable to provide more details. I assume you're using this set up in production?

For now I changed the version to 5.7, as it solves the problems with 5.6.

corro replied:
I haven't heard of problems with MySQL 5.7 when you set `sql_mode` correctly. We have some installations running 5.7, but most are still on 5.5. We plan to update all our installations in the coming months, though.

corro replied:
@gpkvt I just saw that the newest officially supported MySQL version for ILIAS is 5.6. I guess the recommended version should also be 5.6 then. Sorry for the confusion.

gpkvt replied:
@corro No problem. Anyway I changed it back to 5.5 to avoid the problems with 5.6.

Should I add something about `sql_mode` with 5.7 as I did for 5.6?

chfsx replied:
I think a hint about deactivating STRICT_MODE in 5.7 would be nice

gpkvt replied:
@chfsx [x] done

corro commented on docs/INSTALL.md:
````diff
@@ -0,0 +1,659 @@
+git checkout release_5-2
+chown www-data:www-data /var/www/html/ilias -R
+```
+The files MUST be owned by your webserver user/group (e.g. ```www-data``` or ```apache```) the mode SHOULD be 644 for files and 755 for directories.
````
It is bad practice to set the PHP files to be writable by the web user. This allows an attacker to modify the source files if he gains control over the web server processes. To limit the possible damage, the PHP files should be owned by root wherever possible. The only files and directories that must be owned by the web user are:
* ilias.ini.php
* data/

All the other files and directories should be owned by root, but readable by the web user (e.g. 644/755). The simplest way to achieve that, is to remove the `--recursive` (or `-R`) flag from the `chown` command above. This will allow the setup to create/modify the necessary files and leaves the ownership of the rest as is (root).

gpkvt replied:
I moved this to the security section and changed MUST to SHOULD. I think this way we are save and some unexperienced admin doesn't run into strange errors he didn't understand.

What settings do you recommend for the data dir outside the webservers docroot?

corro replied:
@gpkvt the data dir outside of the document root must be read- and writable by the web user, like it says on line 143. No security concerns here ;)

corro commented on docs/INSTALL.md:
```diff
@@ -0,0 +1,659 @@
+
+ILIAS can generate a proper configuration file via the Administration menu ("Administration -> General Settings -> Server -> Java-Server -> Create Configuration File"). Please note that the configuration file is not directly written to the file system, you MUST copy the displayed content and create the file manually.
+
+You MAY use the following SysV-Initscript to start the RPC server:
```
An SystemD configuration file would be nice, since the recommended Linux distributions use SystemD now. Unfortunately I have no such config file at hand, but that should be rather simple (and much more concise).

gpkvt replied:
A minimal ilserver.service for systemd MAY look like this:

```
[Unit]
Description=ILIAS RPC Server
After=network.target

[Service]
ExecStart=/usr/bin/java -Dfile.encoding=UTF-8 -jar /var/www/html/ilias/Services/WebServices/RPC/lib/ilServer.jar /var/www/ilias_data/ilServer.ini start

[Install]
WantedBy=multi-user.target
```

This works fine for start/restart. On stop the process gets stopped but entering the failed state. This also happens if you use:

```
ExecStop=/usr/bin/java -Dfile.encoding=UTF-8 -jar /var/www/html/ilias/Services/WebServices/RPC/lib/ilServer.jar /var/www/ilias_data/ilServer.ini stop
```
Prefixing the ExecStop command with ```-``` (ExecStop=-/usr/bin/java) doesn't worked for me either.

corro replied:
@gpkvt try using the dash (`-`) as the prefix for ExecStart, not ExecStop. It is actually the main process that exists with a non-zero exit code. This systemd unit file works fine in my test environment (using environment variables for common settings):
```
[Unit]
Description=ILIAS RPC Server
After=network.target

[Service]
Environment=JAVA_OPTS="-Dfile.encoding=UTF-8"
Environment=ILSERVER_JAR="/var/www/html/ilias/Services/WebServices/RPC/lib/ilServer.jar"
Environment=ILSERVER_INI="/var/www/html/ilias/Services/WebServices/RPC/lib/ilServer.properties"

ExecStart=-/usr/bin/java $JAVA_OPTS -jar $ILSERVER_JAR $ILSERVER_INI start
ExecStop=/usr/bin/java $JAVA_OPTS -jar $ILSERVER_JAR $ILSERVER_INI stop

[Install]
WantedBy=multi-user.target
```

gpkvt replied:
@corro Thanks, haven't used systemd much, yet. I've replaced the sysv-Initscript, as it can be found in the RPC documentation as well.

gpkvt committed:
Updated according to @corro PR comments

Recommended MySQL version is now 5.7.
Added section "File Access Rights"

tomaszkolonko commented on docs/INSTALL.md:
````diff
@@ -0,0 +1,674 @@
+esac
+
+exit 0
+```
````
Works perfectly well, but since the Server doesn't do anything yet, maybe a link to the README.md (<YOUR_ILIAS_DIR>/Services/WebServices/RPC/lib/README.txt) would come handy for people who want to install lucene right away.

Something like: If you want to install lucene, follow the instructions in the provided readme (link)...

Otherwise then that, it's a great installation document and should help quite a lot ;)

gpkvt replied:
Thanks for your Feedback. The RPC README.txt is linked to now.

gpkvt committed:
Added link to ./Services/WebServices/RPC/lib/README.txt

gpkvt committed:
Replaced SysV-Initscript with systemd service desc

gpkvt committed:
Minor changes to reflect comments on PR

gpkvt committed:
Removed URL to ILIAS-Wiki (page no longer exists)

gpkvt committed:
Added instructions on sql strict mode (5.6+)

gpkvt committed:
Update INSTALL.md

matthiaskunkel commented:
Jour Fixe, April 10, 2017: We highly appreciate this PR and accept it. Thanks to all who contributed to the document. Please change the last sentence of the first paragraph as follows: "ILIAS is published by ILIAS open source e-Learning e.V. under the General Public Licence and free of charge."

gpkvt committed:
Update INSTALL.md

chfsx commented:
For everybody who contributed, we now linked the document in ilias.de: http://www.ilias.de/docu/goto_docu_pg_75024_367.html

Thanks a lot for the work!

gpkvt commented:
@chfsx Thanks. Is there a chance to get proper tables? At least the cellspacing should be increased. Otherwise the version numbers (e.g. MySQL in "Upgrading Dependencies") are hard to read.

chfsx commented:
@gpvt should be possible, I'll update the css of the plugin to have some proper tables. thanks for the hin!

chfsx commented:
@gpkvt I updatet the CSS a bit: http://www.ilias.de/docu/goto_docu_pg_75024_367.html
